crates/solvers/src/domain/solution.rs

```diff
@@ -299,15 +295,15 @@ pub enum Fee {
     Protocol,
 
     /// An additional surplus fee that is charged by the solver.
-    Surplus(U256),
+    Surplus(eth::SellTokenAmount),
 }
 
 impl Fee {
     /// Returns the dynamic component for the fee.
     pub fn surplus(&self) -> Option<U256> {
         match self {
             Fee::Protocol => None,
-            Fee::Surplus(fee) => Some(*fee),
+            Fee::Surplus(fee) => Some(fee.0),
```

Comment:
If the `derive_more` is added to `SellTokenAmount, here we could do:

```suggestion
Fee::Surplus(fee) => Some(fee.into()),
```

Reply:
Would have to be `Some((*fee).into())`, so I just kept it as is for now.